### PR TITLE
Fix label text overflowing

### PR DESCRIPTION
**Pull request by jboolean**
The version I hacked together before didn't actually work.

Use an actual HTML element inside the SVG and take advantage of CSS.

### CODE REVIEW

File: lib/viewcomponents/vertex.jsx [hunk 1]
```diff
@@ -41,20 +41,29 @@ var Vertex = React.createClass({
...
           <path d={roundedRectanglePath(0, 0, vertexWidth, 5, borderRadius, borderRadius, 0, 0)} className="color-bar" fill={styles.get('color')}/>
-          <text ref="titleText" className='label' textAnchor='start' x={padding} y={titlePosition}>
-            {/*model.get('id')*/}
-          </text>
+          {
+            React.createElement('foreignObject', {
+              x: padding,
+              y: titlePosition,
+              width: titleWidth,
+              height: '20',
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Not the height of the vertex?

Comment (jboolean) [inline lib/viewcomponents/vertex.jsx]: Idk. Do you want it to wrap or ellipsis? I'm not sure I know how to make the port labels avoid the vertex label if it is arbitrary height, but that's a minor problem.

**Comment by jboolean** [comment]
Silence is consent.

File: lib/viewcomponents/vertex.jsx (hunk 1 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Well, if you prefer an ellipsis, then `height` should be the height of the text, perhaps, not `'20'`.

Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: But I don't know how you'd check that in a sane way.

File: lib/viewcomponents/vertex.jsx [hunk 2]
```diff
@@ -41,20 +41,29 @@ var Vertex = React.createClass({
...
+              requiredExtensions: 'http://www.w3.org/1999/xhtml'
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Is this how you're getting a div inside of an SVG?

Comment (jboolean) [inline lib/viewcomponents/vertex.jsx]: Yep

**Pull request merged**